File: 옥세훈/20241129_프그_신고결과받기.py
def solution(id_list, report, k):
    answer = []
    dic = dict()
    ans = dict()
    
    for i in id_list:
        dic[i] = []
        
    for i in report:
        a, b = i.split(" ")
        
        if b not in dic[a]:
            dic[a].append(b)
            ans[a] = 0
   
    # print(dic)
    for val in dic.values():
        if val:
            for i in val:
                if i not in ans:
                    ans[i] = 1
                else: 
                    ans[i] += 1

        
    for val in dic.values():
        cnt = 0
        if val:
            for i in val:
                if ans[i] >= k:
                    cnt += 1
                    
            answer.append(cnt)
        else:
            answer.append(0)
    return answer

File: 옥세훈/20241129_프그_신고결과받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        
        HashMap<String, HashSet<String>> map = new HashMap<>();
        HashMap<String, Integer> mapIdx = new HashMap<>();
        
        for (int i=0; i<id_list.length; i++){
            String name = id_list[i];
            map.put(name, new HashSet<>());
            mapIdx.put(name, i);
        }
        
        for (String s : report){
            String[] str = s.split(" ");
            String from = str[0];
            String to = str[1];
            map.get(to).add(from);
        }
        
        for(int i=0; i<id_list.length; i++){
            HashSet<String> send = map.get(id_list[i]);
            if (send.size() >= k) {
                for (String name: send) {
                    answer[mapIdx.get(name)] ++;
                }
            }
        }
        
        
        
        return answer;
    }
}